opus/application/static_media/js/browse.js

```diff
@@ -1196,6 +1200,53 @@ var o_browse = {
         return url;
     },
 
+    // check the cache of both rendered elements and variable o_browse.galleryData; remove 'far away' observations
+    // from cache to avoid buildup of too much data in the browser which slows things down
+    // Two functions; one to delete single elements, just a tool for the main one, manageObservationCache, to loop.
+    manageObservationCache: function(count, append) {
+        let tab = `#${opus.prefs.view}`;
+        var galleryObsElem =  $(`${tab} .gallery [data-obs]`);
+        var tableObsElem = $(`${tab} .op-dataTable-view [data-obs]`);
+
+        function deleteCachedObservation(index) {
+            // don't delete the metadata if the observation is in the cart
+            if (!galleryObsElem.eq(index).hasClass("in")) {
+                let delOpusId = galleryObsElem.eq(index).data("id");
+                delete o_browse.galleryData[delOpusId];
+            }
+            galleryObsElem.eq(index).remove();
+            tableObsElem.eq(index).remove();
+        };
+
+        let beginningIndex = galleryObsElem.first().index();
+        let endingIndex = galleryObsElem.last().index();
+        if (beginningIndex < 0 || endingIndex < 0) {
+            // this only happens when there are no elements rendered, so why bother...
+            // probably don't need to check both, but why not...
+            return;
+        }
+
+        let totalObs = endingIndex - beginningIndex;
+        if (totalObs + count > opus.maxBufferSize) {
+            // if we are appending, remove from the top
+            console.log(`before: ${Object.keys(o_browse.galleryData).length}`);
```

Comment:
Remove debugging log.
Might be easier to read by using Math.min instead of the ?: operator.